### PR TITLE
[global] Changed raw "header('Location:') to Friendica's System::externalRedirect()

### PR DESCRIPTION
**Pull request by Quix0r**
Changes:
- rewrote `header('Location: foo');` to `System::externalRedirect()`
- changed double-quotes to single
- no executable permission on these files

Fixed an issue reported in https://github.com/friendica/friendica/issues/11632#issuecomment-1237755268

### CODE REVIEW

File: saml/saml.php [hunk 1]
```diff
@@ -119,10 +120,7 @@ function saml_sso_initiate(App $a, array &$b)
 	$auth = new \OneLogin\Saml2\Auth(saml_settings());
 	$ssoBuiltUrl = $auth->login(null, [], false, false, true);
 	$_SESSION['AuthNRequestID'] = $auth->getLastRequestID();
-	header('Pragma: no-cache');
-	header('Cache-Control: no-cache, must-revalidate');
```
Comment (MrPetovan) [inline saml/saml.php:123]: Please keep the cache control headers, I’m not sure what would be the side-effects of removing them.

File: saml/saml.php [hunk 2]
```diff
@@ -182,13 +180,9 @@ function saml_slo_initiate(App $a, array &$b)
 	}
 
 	$auth = new \OneLogin\Saml2\Auth(saml_settings());
-
 	$sloBuiltUrl = $auth->logout();
 	$_SESSION['LogoutRequestID'] = $auth->getLastRequestID();
-	header('Pragma: no-cache');
-	header('Cache-Control: no-cache, must-revalidate');
```
Comment (MrPetovan) [inline saml/saml.php:189]: Please keep the cache control headers.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: saml/saml.php (hunk 1 above)
Comment (Quix0r) [reply saml/saml.php:123]: Or cannot we move them to the method `System::externalRedirect()`?

Comment (annando) [reply saml/saml.php:123]: Moving the headers to the function could have got other side effects.

Comment (Quix0r) [reply saml/saml.php:123]: Okay, I would like to have this easier for addon developers, like they don't need to care so much by themselves about it. I will add another parameter to the method.

Comment (Quix0r) [reply saml/saml.php:123]: `require_once(__DIR__ . '/vendor/autoload.php');` <-- Can this line be removed?

**Comment by Quix0r** [comment]
This change requires https://github.com/friendica/friendica/pull/11900 to be merged or else this won't add those headers.

**Pull request closed without merging**